### PR TITLE
Added support for Maps

### PR DESCRIPTION
**Pull request by ryantemple**
Had to name method es6Maps to avoid collision.
Would appreciate suggestions of better name.

### CODE REVIEW

File: test/check-types.js [hunk 1]
```diff
@@ -106,6 +106,10 @@
             assert.isTrue(check.instance([], Array));
         });
 
+        test('instance with new Map and Map returns true', function() {
```
Comment (philbooth) [inline test/check-types.js]: Can you wrap this test inside an `if (typeof Map !== undefined)` check? It causes them to fail when run in older environments.

File: test/check-types.js [hunk 2]
```diff
@@ -374,6 +378,31 @@
             }
         });
 
+        test('es6Map function is defined', function() {
+            assert.isFunction(check.es6Map);
+        });
+
+        test('es6Map with map returns true', function(){
```
Comment (philbooth) [inline test/check-types.js]: This test also needs wrapping for older environments.

**Comment by philbooth** [comment]
Yeah, the naming collision is a bummer. I think I'd prefer `isMap` as an alternative.

Longer term, I may refactor all of the predicates for built-in objects with a more generic function for comparing the result of `Object.prototype.toString`, since that is the common idiom. So this may disappear at the next major version bump, just to warn you.

I've made a couple of comments inline about making the tests pass in older browsers / node 0.10, otherwise this looks good for merging.

**Comment by ryantemple** [comment]
Hopefully this should fix the tests, and have renamed to isMap

**Comment by philbooth** [comment]
Hey @ryantemple, sorry I haven't merged this yet. I'll definitely get round to it this week.

**Comment by philbooth** [comment]
Published to npm as `4.3.0`.

**Pull request merged**